data/battle_scripts_1.s

```diff
@@ -423,6 +423,85 @@ gBattleScriptsForMoveEffects::
 	.4byte BattleScript_EffectHitSetEntryHazard       @ EFFECT_HIT_SET_ENTRY_HAZARD
 	.4byte BattleScript_EffectDireClaw                @ EFFECT_DIRE_CLAW
 	.4byte BattleScript_EffectBarbBarrage             @ EFFECT_BARB_BARRAGE
+	.4byte BattleScript_EffectTripleArrows            @ EFFECT_TRIPLE_ARROWS
+	.4byte BattleScript_EffectInfernalParade          @ EFFECT_INFERNAL_PARADE
+	.4byte BattleScript_EffectTakeHeart               @ EFFECT_TAKE_HEART
+	.4byte BattleScript_EffectAxeKick                 @ EFFECT_AXE_KICK
+
+BattleScript_EffectAxeKick::
```

Comment:
I don't disagree, but I couldn't get anything else working.
I tried something like:
```
BattleScript_EffectAxeKick::
	setmoveeffect MOVE_EFFECT_CONFUSION
	seteffectsecondary
	goto BattleScript_EffectRecoilIfMiss
```
But that doesn't work. I performed multiple combinations with `seteffectchance` and such, but the move only behaves like it should with a battle script like the one I wrote.

Reply:
Look at how other moves with secondary effects are handled (eg. Hurricane). Do you see what's wrong with what you tried?

Reply:
No, I honestly don't.

Hurricane sets the move effect and then jumps to `EFFECT_HIT`'s battle script which includes among the average function macros for an offensive move battle script, `seteffectwithchance`.

As I said, I tried many different combos, both including that same battle script function and not. Nothing has worked.

As a matter of fact, I talked about those attemps inside the Discord server.

![firefox_20230416_163807057](https://user-images.githubusercontent.com/4485172/232337872-ee48ae13-6df3-490d-b4ab-421129bacef5.png)

Reply:
> Hurricane sets the move effect and then jumps to EFFECT_HIT's battle script which includes among the average function macros for an offensive move battle script, seteffectwithchance.

And if you follow down BattleScript_EffectRecoilIfMiss, does it not eventually redirect to the same standard damage script as well? All of your attempts are incorrect due to where you were trying to set the confusion effect. In all of your attempts, you tried to set the target confused before the attackcanceler even ran. If you simply run `setmoveeffect` and then redirect to the Jump Kick battle script, eventually it will reach the `seteffectwithchance` from the standard damage script and confuse the target if the RNG checks out.

Reply:
> does it not eventually redirect to the same standard damage script as well?

No, it doesn't. The battle script of `EFFECT_RECOIL_IF_MISS` *(which to clarify, is the effect of High Jump Kick)* uses its own copy of `EFFECT_HIT` battle script and doesn't jump to it because there's many preproc configs performed on it.

It's basically the same thing I did with Axe Kick; to assign it its own copy of `EFFECT_HIT`'s battle script :eyes:

> If you simply run setmoveeffect and then redirect to the Jump Kick battle script, eventually it will reach the seteffectwithchance from the standard damage script and confuse the target if the RNG checks out.

I did try that too though. No dice.

```
BattleScript_EffectAxeKick::
	setmoveeffect MOVE_EFFECT_CONFUSION
	goto BattleScript_EffectRecoilIfMiss
```
If I set Axe Kick's `.secondaryEffectChance = 100,`, the "inflict confusion on the foe" part of the effect never really happens.

Reply:
> uses its own copy of EFFECT_HIT battle script and doesn't jump to the battle script of EFFECT_HIT

Not what I see in that file:
```
BattleScript_EffectRecoilIfMiss::
	attackcanceler
	accuracycheck BattleScript_MoveMissedDoDamage, ACC_CURR_MOVE
.if B_CRASH_IF_TARGET_IMMUNE >= GEN_4
	typecalc
	jumpifhalfword CMP_COMMON_BITS, gMoveResultFlags, MOVE_RESULT_DOESNT_AFFECT_FOE, BattleScript_MoveMissedDoDamage
.endif
	goto BattleScript_HitFromAtkString
```

The last line redirects to `BattleScript_HitFromAtkString` which is part of the `EFFECT_HIT` battlescript.

---

> BattleScript_EffectAxeKick::
	setmoveeffect MOVE_EFFECT_CONFUSION
	goto BattleScript_EffectRecoilIfMiss

Yup, this is exactly what you need to do. If this isn't working, you need to figure out why the move effect is getting wiped before it reaches `seteffectwithchance`.

Reply:
I forgot to revert the change, because I honestly have no idea how to handle this.

Reply:
> BattleScript_EffectAxeKick::
setmoveeffect MOVE_EFFECT_CONFUSION
goto BattleScript_EffectRecoilIfMiss

If this isn't working, then there's a bug somewhere. You need to go battle script command by battle script command until you figure out where the move effect is getting wiped from memory before it can activate.

Reply:
I understood the first time you said it, but I don't think that's something I can handle.

Reply:
Good news, for whatever reason things just werk now 😂